### PR TITLE
Adding atmospheric pressure forcing

### PR DESCRIPTION
**Pull request by CyprienBosserelle**
- [x] Add/fix pressure forcing
 - [x] Fix Kurganov engine for Pressure forcing
 - [x] Add flags for switching engine

### CODE REVIEW

File: src/Param.h [hunk 1]
```diff
@@ -42,6 +42,8 @@ class Param {
 
 	int doubleprecision = 0; // 0: float precision, 1: double precision
 
+	int engine = 1; // 1: Buttinger, 2: Kurganov, 3: KurganovATMP
```
Comment (AliceHarang) [inline src/Param.h:45]: Add paper reference ?

**Review by AliceHarang — APPROVED** [review]
Nice developments and quite a lot of warning fixed!
I think it would be great to put the ref paper in the model choice (param.h file) and also add a quick simple testcase with an atm forcing (and also testing the "deformation" in the same one?)

File: src/Param.h (hunk 1 above)
Comment (CyprienBosserelle) [reply src/Param.h]: Ok I updated the description with a citing reference. Full reference is in the function documentation.

**Comment by CyprienBosserelle** [comment]
Since there is a lot of fix in this branch many unrelated to atm pressure i will merge this to Dev but add a task to make a test case fro deform and atmp forcing.

**Pull request merged**